### DISCUSSION TITLE
show which agent is being used for a workflow in the UI

### DISCUSSION THREAD

**Discussion by wez** [woodpecker-ci/woodpecker#4222] [Request a Feature]
I quite often find myself wanting to know where a workflow is or was running, especially if it seems to be taking a long time to run, or, when troubleshooting flakeyness that turns out to be environmental.

It would be super convenient if the UI could show the agent name once a given workflow has been picked up.

**Comment by qwerty287** [comment]
Do you mean the queue UI? Then we have this request already in https://github.com/woodpecker-ci/woodpecker/discussions/2162#discussioncomment-8193225

**Reply by wez** [reply]
on the pipeline page, so that you can see the timing of the pipeline steps and understand which agent it was running on 

![image](https://github.com/user-attachments/assets/790ded0f-fb09-4c5a-9ac1-d3261a47e638)

I've discovered that I can `echo $CI_MACHINE` in a step to see the agent name: that appears to be an undocumented variable (not listed on https://woodpecker-ci.org/docs/usage/environment), but it would be great to see at a glance without having to modify every workflow for this.

**Reply by pat-s** [reply]
If you mean to say it should be shown as a label as in the screenshots: these only show for matrix builds and are not the right place for this. For single builds, only commit and branch are shown. 

<img width="183" alt="image" src="https://github.com/user-attachments/assets/cbb2b52d-a3a1-4a00-ad69-f65f9df9c7bc" />

And there is not really much more space left there to possibly show this (arguably optional) information. 

Looking at it, the most suitable idea that comes to my mind is adding another "metadata" tab next to "config/tasks" which holds such information?

**Reply by qwerty287** [reply]
I'd put such info into the debug tab... That's also the reason why you'll probably need it - to debug your pipeline

**Comment by zc-devs** [comment]
> wanting to know where a workflow is or was running
> It would be super convenient if the UI could show the agent name once a given workflow has been picked up.

Depends on the backend, I suppose.

**Comment by pat-s** [comment]
This is now available in [Crow v3.4.0](https://codeberg.org/crowci/crow/releases).
